### ISSUE TITLE
`batch_norm` (training): catastrophic variance cancellation (`E[x²] − E[x]²`) → NaN / wrong output for large-mean + low-variance inputs

### ISSUE THREAD

**Issue by sshonTT** [tenstorrent/tt-metal#45968]
### Component / Area

ops

### Issue Type (optional)

Bad Outputs

### Observed

`ttnn.batch_norm(..., training=True)` on a tensor whose per-channel mean is large relative to its per-channel std produces **NaN / un-normalized** output. 
Minimal case below (`[2, 64, 32, 32]`, per-channel mean ≈ ±5, std ≈ 0.1):

```
[BUG  mean~5  std=0.1]  comp_pcc passed=False  0.7151   tt_std=nan   (cpu_bn std=1.000)
```

### Root cause
The training branch computes the batch variance as **`E[x²] − E[x]²`** (`batch_norm.cpp`, training branch):

```cpp
batch_mean = mean_NHW(input, ...);
auto mean_sq = mean_NHW(square(input), ...);
batch_var = subtract(mean_sq, square(batch_mean), ...);   // E[x²] − E[x]²
```

For a large mean `μ` and small variance, `E[x²]` and `E[x]²` are both ≈ `μ²`, so in bf16 their subtraction **cancels catastrophically** → `batch_var ≈ 0` or slightly negative → `1/sqrt(batch_var + eps)` blows up (huge std) or = **NaN**.

### Suggested fix
Numerically-stable two-pass variance `E[(x − mean)²]`:

```cpp
batch_mean = mean_NHW(input, ...);
auto centered = subtract(input, batch_mean, ...);          // broadcast
batch_var = mean_NHW(square(centered), ...);               // E[(x − mean)²]
```

### Expected

Per-channel normalized output (`std ≈ 1.0`, `PCC ≈ 1.0`), as for a healthy input:

```
[OK   mean~0  std=1.0]  comp_pcc passed=True   0.99999  tt_std=1.000
```

### 1. Steps (exact commands)

branch : https://github.com/tenstorrent/tt-metal/tree/sshon/repro-batchnorm-low-pcc
command : python ./batch_norm_lowvar_repro.py

### 2. Input data / link or description

N/A

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 24.04
Latest tt-metal main branch

### 2. Hardware Details

Blackhole (possibly wh as well), single chip 

### Is this a regression?

Unknown

### Priority

P1

### Impact

Blocks **Gemma-4 (E4B) audio on tt-xla**: `nn.LayerNorm` / `Gemma4RMSNorm`

**Comment by peter941221** [comment]
Opened a PR for this:

https://github.com/tenstorrent/tt-metal/pull/46159

The patch keeps the change narrow:
1. switch the training-path variance from `E[x^2] - E[x]^2` to `E[(x - mean)^2]`
2. add a focused regression for the large-mean, low-variance input shape from this issue

At the time I posted this, I had not completed a hardware-backed rerun in my current environment.
